maintain: allow create grants to take usernames

pdevine opened the pull request:
## Summary

This change fixes the `POST /api/grants` endpoint to allow it to take a user name for the subject similar to how the `PATCH /api/grants` endpoint works. Currently the endpoint will fail incorrectly with a 500 error.


## Checklist

<!-- 
Checklists help us remember things. Change [ ] to [x] to show completion.
-->

- [x] Wrote appropriate unit tests
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)

jmorganca reviewed (approved):
Nice fix 😃

BruceMacD reviewed (commented):
This means creating a grant by group name will work also, correct?

dnephin reviewed (approved):
Looks good!

Edit: Nevermind! RequireOneOf is already exactly one!

pdevine commented:
@BruceMacD Yes! I've added a test for that, and realized there was a bug with group names where it was not creating the correct ID type. That is fixed now too (and the unit test is also fixed)
